crates/saya/core/src/lib.rs

```diff
@@ -341,63 +447,101 @@ impl Saya {
     ///
     /// * `prove_scheduler` - A full parallel prove scheduler.
     /// * `last_block` - The last block number in the `prove_scheduler`.
-    async fn process_proven(&self, prove_scheduler: Scheduler) -> SayaResult<()> {
-        // Prove each of the leaf nodes of the recursion tree and merge them into one
-        let (proof, state_diff, (_, last_block)) =
-            prove_scheduler.proved().await.context("Failed to prove.")?;
-
+    async fn process_proven(
+        &mut self,
+        proof: ProverResult,
+        state_diff: Vec<Felt>,
+        last_block: u64,
+        state_roots: (Felt, Felt),
+    ) -> SayaResult<()> {
         trace!(target: LOG_TARGET, last_block, "Processing proven blocks.");
-
-        if self.config.store_proofs {
-            let filename = format!("proof_{}.json", last_block);
-            let mut file = File::create(filename).await.context("Failed to create proof file.")?;
-            file.write_all(proof.as_bytes()).await.context("Failed to write proof.")?;
-        }
-
-        let serialized_proof: Vec<FieldElement> = parse(&proof)?.into();
-        let world_da = state_diff.world_da.unwrap();
-
-        // Publish state difference if DA client is available
+        let serialized_proof = proof.serialized_proof;
+        // Publish state difference if DA client is available.
         if let Some(da) = &self.da_client {
             trace!(target: LOG_TARGET, last_block, "Publishing DA.");
 
-            if self.config.skip_publishing_proof {
-                da.publish_state_diff_felts(&world_da).await?;
+            let checkpoint = PublishedStateDiff {
+                prev_state_root: state_roots.0,
+                state_root: state_roots.1,
+                prev_height: self.prev_height,
+                prev_commitment: self.prev_commitment,
+                proof: serde_json::to_value(&proof.proof).unwrap(),
+            };
+            // let ns = Namespace::new_v0(b"saya-dev").unwrap();
+            // let commitment = Commitment::from_blob(ns, 0, &serialized_proof.iter().map(|felt| felt.to_bytes()).collect::<Vec<_>>());
+            let (commitment, height) = if self.config.mode != SayaMode::Ephemeral {
+                da.publish_checkpoint(checkpoint).await?
+            } else if self.config.skip_publishing_proof {
+                da.publish_state_diff_felts(&state_diff).await?
             } else {
-                da.publish_state_diff_and_proof_felts(&world_da, &serialized_proof).await?;
-            }
+                da.publish_state_diff_and_proof_felts(&state_diff, &serialized_proof).await?
+            };
+            self.prev_commitment = Some(commitment);
+            self.prev_height = Some(height);
+
+            info!(target: LOG_TARGET,"commitment: {:?}, height: {:?}", commitment.0, height);
```

Comment:
If we have a DA configured, let's return. We don't need to post on Starknet for now.